tests/test_filters.py

```diff
@@ -29,6 +39,14 @@ async def test_equal(self):
             set(await CharFields.filter(char="moo").values_list("char", flat=True)), {"moo"}
         )
 
+    async def test_enum(self):
+        self.assertEqual(
+            set(await CharFields.filter(char=MyEnum.moo).values_list("char", flat=True)), {"moo"}
```

Comment:
I don't think this is a correct test case. `char` is a str field, why should it be possible to search by an Enum.

Reply:
I try it with Django:
```bash
mkdir something
cd something
python3.11 -m venv venv
source venv/bin/activate
pip install django ipython
python -m django startproject something .
python manage.py startapp foo
echo '
from enum import Enum, StrEnum
from django.db import models

class RawEnum(Enum):
    moo = "moo"

class MyEnum(str, Enum):
    moo = "moo"

class MyStrEnum(StrEnum):
    moo = "moo"

class MyModel(models.Model):
    name = models.CharField(max_length=10)
' > foo/models.py
python -c 's="INSTALLED_APPS = [\n";m="foo";from pathlib import Path;p=Path("something/settings.py");p.write_text(p.read_text().replace(s,s.strip()+repr(m)+",\n"))'
python manage.py makemigrations
python manage.py migrate
python manage.py shell
```
Got:
```
In [1]: from foo.models import *

In [2]: MyModel.objects.create(name='moo')
Out[2]: <MyModel: MyModel object (1)>

In [3]: MyModel.objects.filter(name=RawEnum.moo)
Out[3]: <QuerySet []>

In [4]: MyModel.objects.filter(name=MyEnum.moo)
Out[4]: <QuerySet [<MyModel: MyModel object (1)>]>

In [5]: MyModel.objects.filter(name=MyStrEnum.moo)
Out[5]: <QuerySet [<MyModel: MyModel object (1)>]>
```
I think we can follow the strategy of django:
1. Return empty queryset if the enum object only inherit from Enum (result of `.filter(name=str(enum_obj))`)
2. Use its value to query if the enum object inherit from str and Enum

Reply:
Does Django have a test case or documentation explicitly stating that they support filtering `CharField` with enums? I'm wondering if the ability to find records in your examples is just a weird coincidence.

If django doesn't explicitly support it, I don't think we should either. I think in the ideal world `MyModel.objects.filter(name=RawEnum.moo)` should cause a type error because you are trying to filter by incompatible types. Having a test case for this kinda make this behavior official which I don't think we should be doing. I would suggest making this PR not about searching by enum but about updating pypika-tortoise version.

Reply:
Yeah, that's a good point, I agree!